runtime/src/chpl-launcher-common.c

```diff
@@ -102,6 +103,42 @@ void chpl_append_to_largv(int* largc, const char*** largv, int* largv_len,
   (*largv)[(*largc)++] = (arg);
 }
 
+// Helper for appending arguments to a variable-size command buffer.
+// - Requires the buffer pointer is uninitialized on first call.
+// - After exceeding an initial allocated size estimate, each call will allocate
+//   additional memory as needed.
+char* chpl_append_to_cmd(char* cmdBuf, const char* format, ...) {
```

Comment:
I also suspect you may need `__attribute__ ((format (printf, 2, 3)))` on this function, but I could be wrong